lib/erasure/erasure.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package erasure
+
+import (
+	"errors"
+
+	"github.com/klauspost/reedsolomon"
+)
+
+// ErrNotEnoughValidators cannot encode something for zero or one validator
+var ErrNotEnoughValidators = errors.New("expected at least 2 validators")
+
+// ObtainChunks obtains erasure-coded chunks, divides data into number of validatorsQty chunks and
+// creates parity chunks for reconstruction
+func ObtainChunks(validatorsQty int, data []byte) ([][]byte, error) {
+	recoveryThres, err := recoveryThreshold(validatorsQty)
+	if err != nil {
+		return nil, err
+	}
+	enc, err := reedsolomon.New(validatorsQty, recoveryThres)
+	if err != nil {
+		return nil, err
+	}
+	shards, err := enc.Split(data)
+	if err != nil {
+		return nil, err
+	}
+	err = enc.Encode(shards)
+	if err != nil {
+		return nil, err
+	}
+
+	return shards, nil
+}
+
+// Reconstruct decodable data from a set of chunks
+func Reconstruct(validatorsQty int, chunks [][]byte) error {
```

Comment:
As part of Reconstruct we should get the original `data` back. It looks like we might be able to do that using `enc.Join`.
In the end return `[]byte, error`

Reply:
added return to get original data back.